### ISSUE TITLE
shib / shib_path not uninstalling on config:import 

### ISSUE THREAD

**Issue by druliner** [jvogtuw/shibboleth#8]
In Drupal 10
shibboleth and shibboleth_path ALWAYS install on config:import (cim) even if they're not present in the core.extension.yml

drush @site.name cim
 [notice] Synchronized extensions: install shibboleth.
 [notice] Synchronized extensions: install shibboleth_path.

If you DISABLE the module manually or with drush and do a export config (cex) it continues to say that it's removing  shib & shib_path (eg: it never says "The active configuration is identical to the configuration in the export directory")

drush @site.name pm:uninstall shibboleth shibboleth_path
drush @site.name cex --diff
output: 
 - shibboleth: 0
 - shibboleth_path: 0
 drush @site.name cex --diff
output: 
 - shibboleth: 0
 - shibboleth_path: 0
...

This was much more annoying before you fixed the error in shibboleth_path because after a config:import I'd immediately have to do a pm:uninstall shibboleth_path or the site would crash. D: 

Thanks again for this module and for fixing the shibboleth_path error in D10 today! later

**Comment by jvogtuw** [comment]
Well that's...weird. I'll check it out.

**Comment by jvogtuw** [comment]
Do you have config_ignore installed? I was having issues with cex with config_ignore installed on a D9 -> D10 migrated site. It was supposed to ignore *:uuid, but something was going wrong. cex would export everything with the uuid moved to the bottom and that was still getting picked up as a change. So no matter how many times I ran cim and cex, every config was listed as changed between active and staged. Which sounds similar to what you're experiencing.

Uninstalling and reinstalling config_ignore (8.x-3.1) seems to have resolved that issue and I can't replicate this shib issue.

**Comment by jvogtuw** [comment]
If you are using config_ignore 3.1 try setting it to Intermediate and put your things to ignore in the Import field and leave the Export field blank. This is what I had to do to get it to actually ignore UUIDs if the exported config already had it.

**Comment by druliner** [comment]
Yeah. I'm not using config_ignore. I'll spin up a fresh site with nothing but shibboleth installed and see if I can repo there. In retrospect I should have done that in the first place. Happy Friday Eve!

**Comment by jvogtuw** [comment]
Did spinning up a new site solve the issue? I haven't encountered this specific problem, but I was looking over the code a realized there's no schema for shibboleth_path.settings and an incomplete one of shibboleth.settings. I think the schemas are now required for D10+.

**Comment by druliner** [comment]
I’ll try and check today.

More than likely I screwed something up. I’m still getting my head around the config system.

Happy Friday!
dd

**Comment by jvogtuw** [comment]
No rush! Lemme know if you do ever come across it again.

**Comment by jvogtuw** [comment]
Happy 2026! I'm going to assume the problem didn't resurface and close this issue. Feel free to reopen if it does.
